### DISCUSSION TITLE
Sharing users between FediVerse/ActivityPub services

### DISCUSSION THREAD

**Discussion by 0xlord** [mastodon/mastodon#17139] [Q&A] [answered]
Hello, 

Just learned about Mastodon, FediVerse, and ActivityPub in General. But I can't find if users can be shared between instances. I don't necessarily talk between domains, but inside one. An example:

Let's say I have the domain `example.net`. Now I want to host 
 * Mastodon as `toot.example.net`
 * Pixelfed as `pics.example.net`
 * Mobilizon as `events.example.net`

If I want to use each of them I'd end up, to my understanding, with three users:
 * `0xlord@toot.example.net`
 * `0xlord@pics.example.net`
 * `0xlord@events..example.net`

Is there a way to have one domain handling the users and giving them access to each service? For example registering on `pics.example.net` would register `0xlord@example.net` and putting it up between all three services. In the best case to link follows/followers and activities between them? For example:
I put up an Event at Mobilizon. It automatically put down a Toot with the link to it on Mastodon.

Would something like this be possible? Googled for it for hours. Yet my Google-Fu doesn't help me here ...

**Comment by trwnh** [comment, marked as answer]
This would only be possible if you had a generic ActivityPub Server and every service you wanted to use acted as an ActivityPub Client. Because the three apps you mentioned are monolithic (like having their own integrated server), you will need three actors.

**Reply by 0xlord** [reply]
This is really helping. Thank you very much. Looking this up and experiment a bit with it. Coming back if I have additional questions if you don't mind.

**Reply by Gargron** [reply]
You could have an application that would handle the ActivityPub base layer (actors, inbox, outbox, and so on) and installable extensions on top that would provide the business logic for different use cases, without requiring ActivityPub C2S. That being said, that's not what Mastodon is or intends to be.
